jlh/WeatherApp/app/src/androidTest/java/com/e_conomic/weatherapp/SimpleInstrumentationTest.kt

```diff
@@ -0,0 +1,58 @@
+package com.e_conomic.weatherapp
+
+import com.e_conomic.weatherapp.ui.activities.MainActivity
+import android.support.test.rule.ActivityTestRule
+import android.support.test.espresso.Espresso.*
+import android.support.test.espresso.matcher.ViewMatchers.withId
+import android.support.test.espresso.action.ViewActions.click
+import android.support.test.espresso.action.ViewActions.replaceText
+import android.support.test.espresso.contrib.RecyclerViewActions
+import android.support.test.espresso.assertion.ViewAssertions.matches
+import android.support.test.espresso.matcher.BoundedMatcher
+import android.support.test.espresso.matcher.ViewMatchers.*
+import android.support.v7.widget.RecyclerView
+import android.support.v7.widget.Toolbar
+import android.widget.TextView
+import org.hamcrest.Description
+import org.hamcrest.Matcher
+import org.hamcrest.Matchers.`is`
+
+import org.junit.Rule
+import org.junit.Test
+
+class SimpleInstrumentationTest {
+
+    @get:Rule val activityRule = ActivityTestRule(MainActivity::class.java)
```

Comment:
It works like the usual java JUnit rules, but since there are no fields in Kotlin, only _properties_, JUnit would not work if we just put `@Rule`. Therefore we have to annotate the getter instead, which is what this does.

I think you could also do something like `@Rule @JvmField` which exposes the backing field of the property :)